src/libs/vehicle/ardupilot/ardupilot.ts

```diff
@@ -298,6 +300,8 @@ export abstract class ArduPilotVehicle<Modes> extends Vehicle.AbstractVehicle<Mo
 
         this._isArmed = Boolean(heartbeat.base_mode.bits & MavModeFlag.MAV_MODE_FLAG_SAFETY_ARMED)
         this.onArm.emit()
+        this._flying = heartbeat.system_status.type === MavState.MAV_STATE_ACTIVE
```

Comment:
Is this MAV_STATE_ACTIVE shared between all vehicles? @ES-Alexander maybe can help answering that one.

If so, maybe we can have a better name that also works for subs/boats/rovers, but if we don't find one, flying is also ok. I usually say or subs are flying 😅